src/components/Control.js

```diff
@@ -53,20 +54,24 @@ export const css = ({
 });
 
 const Control = (props: ControlProps) => {
-  const { children, cx, getStyles, className, isDisabled, isFocused, innerRef, innerProps, menuIsOpen, emotion } = props;
+  const { children, cx, getStyles, className, isDisabled, isFocused, innerRef, innerProps, menuIsOpen } = props;
   return (
-    <div
-      ref={innerRef}
-      className={cx(emotion.css(getStyles('control', props)), {
-        'control': true,
-        'control--is-disabled': isDisabled,
-        'control--is-focused': isFocused,
-        'control--menu-is-open': menuIsOpen
-      }, className)}
-      {...innerProps}
-    >
-      {children}
-    </div>
+    <ClassNames>
+      {({ css }) => (
+        <div
+          ref={innerRef}
+          className={cx(css(getStyles('control', props)), {
```

Comment:
Here we're using the ClassNames fn from emotion 10 to access the css method so we can pass the processed style objects from css(getStyles()) to our custom cx implementation. 

@mitchellhamilton I just wanted to sanity check the fact that we're using a custom cx implementation her instead of the one provided to us by ClassNames is ok.

Reply:
Using your own cx function is totally fine, it just means that emotion styles won't be merged together which in this case, I don't believe it matters. I'm not really sure about this whole approach though, ClassNames is intended to be a workaround when a component doesn't accept a className prop, I'd recommend an approach like this:
```diff
diff --git a/src/components/Control.js b/src/components/Control.js
index a964547..c38a9ad 100644
--- a/src/components/Control.js
+++ b/src/components/Control.js
@@ -1,6 +1,7 @@
 // @flow
-import React, { type Node, type ElementRef } from 'react';
-import { ClassNames } from '@emotion/core';
+import { type Node, type ElementRef } from 'react';
+/** @jsx jsx */
+import { jsx } from '@emotion/core';

 import type { CommonProps, PropsWithStyles } from '../types';

@@ -56,22 +57,19 @@ export const css = ({
 const Control = (props: ControlProps) => {
   const { children, cx, getStyles, className, isDisabled, isFocused, innerRef, innerProps, menuIsOpen } = props;
   return (
-    <ClassNames>
-      {({ css }) => (
-        <div
-          ref={innerRef}
-          className={cx(css(getStyles('control', props)), {
-            'control': true,
-            'control--is-disabled': isDisabled,
-            'control--is-focused': isFocused,
-            'control--menu-is-open': menuIsOpen
-          }, className)}
-          {...innerProps}
-        >
-          {children}
-        </div>
-      )}
-    </ClassNames>
+    <div
+      ref={innerRef}
+      css={getStyles('control', props)}
+      className={cx({
+        'control': true,
+        'control--is-disabled': isDisabled,
+        'control--is-focused': isFocused,
+        'control--menu-is-open': menuIsOpen
+      }, className)}
+      {...innerProps}
+    >
+      {children}
+    </div>
   );
 };

diff --git a/src/utils.js b/src/utils.js
index 0787bfd..d37f037 100644
--- a/src/utils.js
+++ b/src/utils.js
@@ -41,11 +41,10 @@ function applyPrefixToName(prefix, name) {

 export function classNames(
   prefix?: string | null,
-  cssKey?: string | null,
   state?: ClassNamesState,
   className?: string,
 ) {
-  const arr = [cssKey, className];
+  const arr = [className];
   if (state && prefix) {
     for (let key in state) {
       if (state.hasOwnProperty(key) && state[key]) {
```

Reply:
@mitchellhamilton This is objectively much nicer, only issue is that it seems converting to 
```javascript 
<div css={someemotionstyles} className={react-select-generated-classnames}/>
```
generates the following selector `react-select-classname emotion-classname' 
Its important that we generate the selector as `emotion-classname generated-react-select-classname' as we need to preserve css specificity for the react-select classname  for users not using css-in-js to easily override emotion styles without having to resort to !important. 

If there's a good way around this,, I'd love to refactor the code to the pattern you've specified above, as it's considerably cleaner.